File: src/main/java/fr/lewon/bot/runner/errors/BotOperationExecutionException.java
package fr.lewon.bot.runner.errors;

import fr.lewon.bot.runner.bot.operation.BotOperation;

public class BotOperationExecutionException extends Exception {

    private static final long serialVersionUID = 8125206445264588526L;

    public BotOperationExecutionException(BotOperation botOperation, Exception cause) {
        super("An error occurred when running [" + botOperation.getClass().getCanonicalName() + "]", cause);
    }
}

File: src/main/kotlin/fr/lewon/bot/runner/errors/BotOperationExecutionException.kt
package fr.lewon.bot.runner.errors

import fr.lewon.bot.runner.bot.operation.BotOperation

class BotOperationExecutionException(botOperation: BotOperation, message: String? = null, cause: Exception? = null) :
        Exception("An error occurred when running [" + botOperation.javaClass.canonicalName + "] : $message", cause)
